src/commands/env/list.ts

```diff
@@ -75,13 +75,16 @@ export default class EnvList extends SfCommand<Environments> {
 
     const final: Environments = [];
 
-    try {
-      const results = await SfHook.run(this.config, 'sf:env:list', { all: this.flags.all });
-      const tables = results.successes
-        .map((r) => r.result)
-        .reduce((x, y) => x.concat(y), [])
-        .filter((t) => t.data.length > 0);
+    const results = await SfHook.run(this.config, 'sf:env:list', { all: this.flags.all });
+    const tables = results.successes
+      .map((r) => r.result)
+      .reduce((x, y) => x.concat(y), [])
+      .filter((t) => t.data.length > 0);
 
+    if (tables.length === 0) {
+      cli.log(messages.getMessage('error.NoResultsFound'));
```

Comment:
@mdonnalley should this be guarded with a check for json being enabled?

Reply:
It depends.

Since I used cli-ux to log here, I should check for the json flag first. However, if I use `this.log` instead, then oclif will not log if json is enabled.

Generally speaking, we should use `this.log` instead of `cli-ux` to log to the console for this very reason